test/tablecloth/time/api/slice_test.clj

```diff
@@ -17,76 +15,68 @@
              (index-by :A)
              (slice 2 3)))))
 
+(deftest slice-by-year
+  (is (= (dataset {:A [1979 1980]
+                   :B [9 10]})
+         (-> (dataset {:A (map #(+ % 1970) (range 11))
```

Comment:
Treating years as numbers. This will not be the last time we think about this, but we'll start with this idea.